metomi/rose/popen.py

```diff
@@ -188,9 +188,17 @@ def run(self, *args, **kwargs):
             stdin = kwargs.get("stdin")
         stdout, stderr = proc.communicate(stdin)
         retcode = proc.wait()
-        if isinstance(retcode, bytes):
-            retcode = retcode.decode()
-        return retcode, stdout, stderr
+
+        def _decode_if_bytes(obj):
+            if isinstance(obj, bytes):
+                return obj.decode()
+            return obj
+
+        return (
+            _decode_if_bytes(retcode),
+            _decode_if_bytes(stdout),
+            _decode_if_bytes(stderr)
+        )
```

Comment:
Instead of doing this, why not use `Popen(..., text=True)` below in `def run_bg()`?

https://docs.python.org/3.7/library/subprocess.html#subprocess.run